lib/AST/ProtocolConformance.cpp

```diff
@@ -185,6 +185,65 @@ ProtocolConformanceRef::getWitnessByName(Type type, DeclName name) const {
   return getConcrete()->getWitnessDeclRef(requirement);
 }
 
+
+static bool classifyRequirement(ModuleDecl *module, 
+                         ProtocolConformance *reqConformance, 
+                         ValueDecl *requiredFn) {
+  auto DC = reqConformance->getDeclContext();
+  auto reqTy = reqConformance->getType();
+  auto declRef = reqConformance->getWitnessDeclRef(requiredFn);
+  auto witnessDecl = cast<AbstractFunctionDecl>(declRef.getDecl());
+  switch (witnessDecl->getRethrowingKind()) {
+    case FunctionRethrowingKind::ByConformance:
+      if (module->classifyWitnessAsThrows( 
+        reqTy->getContextSubstitutionMap(module, DC))) {
```

Comment:
Can this just be reqConformance->getSubstitutions(module)?

Reply:
you are the expert w/ these, I am unsure. I can try using that instead in a future iteration; unless you think that is a critical change?